### PR TITLE
Windows events batch 2 v2

### PR DESCRIPTION
**Pull request by yaelrbergman**
-----------------------------------------------------------------------------------------------------------
## Before submitting this PR please ensure that you have read the following sections and filled out the changes, reason for change and testing complete sections:

Thank you for your contribution to the Microsoft Sentinel Github repo.

> Details of the code changes in your submitted PR.  Providing descriptions for pull requests ensures there is context to changes being made and greatly enhances the code review process.  Providing associated Issues that this resolves also easily connects the reason.
   
  Change(s):
Added WindowsEvent (event 4688) source to the following detections:
NetworkEndpointCorrelation
Nobelium_FoggyWeb
SOURGUM_IOC
WSLMalwareCorrelation
base64_encoded_pefile
execute_base64_decodedpayload
HAFNIUMNewUMServiceChildProcess
malware_in_recyclebin
NOBELIUM_SuspiciousRundll32Exec
SuspiciousScriptRegistryWrite
PotentialBuildProcessCompromise
 

 Reason for Change(s):
Added WindowsEvent data in addition to securityEvent

> The code should have been tested in a Microsoft Sentinel environment that does not have any custom parsers, functions or tables, so that you validate no incorrect syntax and execution functions properly.  If your submission requires a custom parser or function, it must be submitted with the PR.

   Testing Completed:
   - manual testing

_Note: If updating a detection, you must update the version field._

> After the submission has been made, please look at the Validation Checks.

   Checked that the validations are passing and have addressed any issues that are present:
   - Yes/No/Need Help
   
   _Note: Let us know if you have tried fixing the validation error and need help._

> References: 
> - [Guidance for Detection checks](https://github.com/Azure/Azure-Sentinel#pull-request-detection-template-structure-validation-check)
> - [General contribution guidance](https://github.com/Azure/Azure-Sentinel/wiki#what-can-you-contribute-and-how-can-you-create-contributions)
> - [PR validation troubleshooting](https://github.com/Azure/Azure-Sentinel#pull-request)

-----------------------------------------------------------------------------------------------------------

### CODE REVIEW

**Review by oshezaf — CHANGES_REQUESTED** [review]
One last question! I promise (well I can never promise)

File: Detections/SecurityEvent/PotentialBuildProcessCompromise.yaml [hunk 1]
```diff
@@ -54,10 +58,47 @@ query: |
...
+  | where Process has_any (build_processes)
+  | extend ParentProcessName = tostring(EventData.ParentProcessName)
+  | extend Account =  strcat(tostring(EventData.SubjectDomainName),"\\", tostring(EventData.SubjectUserName))
+  | extend CommandLine = tostring(EventData.CommandLine) 
+  | summarize by BuildParentProcess=ParentProcessName, BuildProcess=Process, BuildAccount = Account, Computer, BuildCommand=CommandLine, timekey= bin(TimeGenerated, time_window), BuildProcessTime=TimeGenerated
+  | join kind=inner(
+  WindowsEvent
+  | where TimeGenerated > ago(timeframe)
+  // Look for file modifications to code file
+  | where EventID == 4663 and EventData has_any ("0x6", "0x4", "0X100") and EventData has_any ("*.cs$", "*.cpp$")
```
[the remark this answers is not in the document]
Comment (yaelrbergman) [reply Detections/SecurityEvent/PotentialBuildProcessCompromise.yaml:77]: I tried to replace the endswith in line 83 with "$", the regexp won't work here?

**Comment by shainw** [comment]
@yaelrbergman - Please be sure to fill out the PR template, it is required.  Additionally, looks you have conflicts with master that need to resolved.

File: Detections/SecurityEvent/PotentialBuildProcessCompromise.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (oshezaf) [reply Detections/SecurityEvent/PotentialBuildProcessCompromise.yaml:77]: No has_any does not support regex. It does not matter. has_any only prefitler. Line 83 will do the rest of the job. So, remove the "$"

**Review by oshezaf — APPROVED** [review]

**Pull request merged**